lib/pool.js

```diff
@@ -33,7 +33,13 @@ function Pool(url, options = {}) {
   )
 }
 
-Pool.prototype.request = function (method, path, data) {
+Pool.prototype.request = function (method, url, params, data) {
+  if (! url || typeof url.render !== 'function') {
+    return P.reject(new Error('Invalid url argument'))
+  }
+
+  const path = url.render(params)
```

Comment:
Just realised we shouldn't let the `throw` leak out of here uncaught, it needs to reject instead.